reconcile/openshift_base.py

```diff
@@ -73,35 +78,44 @@ def init_specs_to_fetch(ri, oc_map,
                 continue
 
             namespace = namespace_info['name']
+            # These may exit but have a value of None
             managed_resource_names = \
-                namespace_info.get('managedResourceNames')
+                namespace_info.get('managedResourceNames') or []
             managed_resource_type_overrides = \
-                namespace_info.get('managedResourceTypeOverrides')
+                namespace_info.get('managedResourceTypeOverrides') or []
 
             # Initialize current state specs
             for resource_type in managed_types:
                 ri.initialize_resource_type(cluster, namespace, resource_type)
-                # Handle case of specific managed resources
-                resource_names = \
-                    [mrn['resourceNames'] for mrn in managed_resource_names
-                     if mrn['resource'] == resource_type] \
-                    if managed_resource_names else None
-                # Handle case of resource type override
-                resource_type_override = \
-                    [mnto['override'] for mnto
-                     in managed_resource_type_overrides
-                     if mnto['resource'] == resource_type] \
-                    if managed_resource_type_overrides else None
-                # If not None, there is a single element in the list
-                if resource_names:
-                    [resource_names] = resource_names
-                if resource_type_override:
-                    [resource_type_override] = resource_type_override
+            resource_names = {}
+            resource_type_overrides = {}
+            for mrn in managed_resource_names:
+                # Current implementation guarantees only one
+                # managed_resource_name of each managed type
+                if mrn['resource'] in managed_types:
+                    resource_names[mrn['resource']] = mrn['resourceNames']
+                else:
+                    logging.info(
```

Comment:
in this case, we'd like to restore this: https://github.com/app-sre/qontract-reconcile/blob/7e79c0926afb02b7aa5fcaa56213002f86a64c0d/reconcile/openshift_resources_base.py#L564-L575

this should mean that you will not need to make any changes to any integrations, only to the base.

Reply:
Totally. And the change that restores it is 65fa050b247e4f8f5836dfc85dd09c4f42eff8ab . In particular, go down to the changes to `reconcile/utils/openshift_resource.py`. The problem is that the `KeyError` there used to be dropped on the floor, so this `except` block was never reached.

Once I restore it, we have a bunch of additional callers that don't seem to want that exception at all and that risk crashing on it - limit ranges, network policies an so on. That's why I adapt all of them, simply with an `with suppress(KeyError):` block - it's the largest change for which I'm kinda confident I don't need to write unit tests. I agree that it'd be a misconfiguration if there were such an exception, but our current behaviour is to drop it, so in here I'm following suit. 

It'd be very easy to let them crash and get the configs fixed, but if we decide to do so it should be in a different PR.

Finally, there's an extra caveat: this block applies only to resources that are listed under `openshiftResources` in the namespace or cluster.

However, `init_specs_to_fetch` does an earlier filtering of `managedResourceNames` and `managedResourceTypeOverrides`. If a `managedResourceName` is of a type not in `managedResourceTypes` it is currently dropped and we don't even get to generate a `StateSpec` that could be used in `ri.add_current` or `ri.add_desired`. Those have to be logged here.

This one was harder than it looked, I really needed to research the code. ;)